x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/results/CategoryDefinition.java

```diff
@@ -152,6 +167,26 @@ public void setGrokPattern(String grokPattern) {
         this.grokPattern = grokPattern;
     }
 
+    public long[] getPreferredToCategories() {
+        return preferredToCategories;
+    }
+
+    public void setPreferredToCategories(long[] preferredToCategories) {
+        this.preferredToCategories = preferredToCategories;
```

Comment:
Since the value is being serialized using `writeVLongArray()` this should throw if any value in the array is negative.